docs/tuf-spec.txt

```diff
@@ -530,6 +532,14 @@ Version 1.0 (Draft)
            , ... }
      }
 
+   COMPRESSION_ALGORITHM specifies one of the compression algorithms supported
```

Comment:
Listing the compression algorithms used by the repository removes the need to list all compressed versions of metadata in Snapshot, reducing its size and minimizing code complexity.  Should the benefits  of listing the compression algorithm(s) be stated in the specification, or are you simply curious for the change?

There were a few other reasons, which are captured in an email thread with David Lawrence @endophage (see email thread with title, "role names").  I'll quote some part of his request to remove compressed metadata in Snapshot, etc.


> I feel very strongly that data compression should be implemented at layer 6 and TUF should not be validating compressed files. As is, a malicious publisher could intentionally publish decompression bomb. The TUF user would download and validate the hash and size of the compressed file but be vulnerable when they decompress. By implementing compression at layer 6, the client will decompress the stream as it arrives and never read more than Length bytes from the incoming stream.
> 
> If we were talking about GBs of data there would be a strong argument for compression on disk, but TUF data is (and necessarily so) in the order of KB and the storage gains from compression are minimal. Compression at layer 6 still provides the benefit of reduced network transfer.
> 
> ...
> 
> Overall, I think consistently not using file extensions when referencing TUF metadata within the metadata files, and specifying any compression is done at layer 6 overall is a security improvement on 2 fronts. Firstly it makes the metadata more consistent, reducing code complexity, which makes auditing easier and chance of bugs lower. Secondly, it better protects the user against decompression bombs. Note that in the Python implementation the code attempts to read the entire gzipped file and as such is vulnerable to a decompression bomb: utils.py:934: fileobject = six.StringIO(gzip.open(filepath).read().decode('utf-8’))